projects/stackoverflow/src/stackoverflow/DataExtractor.java

```diff
@@ -0,0 +1,178 @@
+package stackoverflow;
+
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.InputStream;
+import java.util.Iterator;
+
+import javax.xml.stream.XMLEventFactory;
+import javax.xml.stream.XMLEventReader;
+import javax.xml.stream.XMLEventWriter;
+import javax.xml.stream.XMLInputFactory;
+import javax.xml.stream.XMLOutputFactory;
+import javax.xml.stream.XMLStreamException;
+import javax.xml.stream.events.Attribute;
+import javax.xml.stream.events.XMLEvent;
+
+public class DataExtractor {
+
+	// paths of the files that we read from and write to
+	private String inputFilePath;
+	private String outputFilePath;
+
+	// the name of the row tag.
+	private static final String ROW = "row";
+	// the name of the posts tag
+	private static final String POSTS = "posts";
+	// some constant strings for the relevant attributes's names in the row tag
+	private static final String ID = "Id";
+	private static final String Post_TYPE_ID = "PostTypeId";
+	private static final String CREATION_DATE = "CreationDate";
+	private static final String OWNER_USER_ID = "OwnerUserId";
```

Comment:
General comment - looks like you're using tabs. Can you equally have used 2 spaces instead, or is it more difficult in some way?